### DISCUSSION TITLE
X-Cache-Status is always Bypass

### DISCUSSION THREAD

**Discussion by raselSoftNode** [Kong/kong#11326] [Help]
i have a problem with kong proxy cache . in my case X-Cache-Status always "Bypass".

this is my server =>
```js
const express=require("express")
const cors=require("cors")

const app=express()

app.get("/echo",(req,res)=>{
    res.setHeader('Cache-Control', 'max-age=300'); 

    res.status(200).json({name:"rasel",age:25,sub:["bangla","english"]})
})

app.get("/remote",(req,res)=>{
    res.setHeader('Cache-Control', 'max-age=300'); 

    res.status(200).json({name:"this is remote server"})
})

app.listen(4000,()=>{
    console.log("http://localhost:4000")
})
```

And this is my kong.yml file => 
```js
_format_version: "2.1"


services:
  - name: echo-server
    host: host.docker.internal
    port: 4000
    path: /echo
    routes:
      - name: localhost_echo_route
        paths:
        - /echo_server
        - /echo_server2
  - name: remote_server
    host: host.docker.internal
    port: 4000
    path: /remote
    routes:
     - name: localhost_remote_routee
       paths:
        - /remote_server
        - /remote_server2

# plugins
plugins:
- name: proxy-cache
  config:
    response_code:
    - 200
    request_method:
    - GET
    - HEAD
    content_type:
    - text/plain
    - application/json
    cache_ttl: 300
    strategy: memory
    cache_control: false

```

My Browser Response => 
![image](https://github.com/globocom/kong-plugin-proxy-cache/assets/126503071/59c4e762-0d98-467d-9257-009b3a1f1f0b)

**Comment by ms2008** [comment]
Because express returns a response header `Content-Type` with a parameter suffix, note the `charset=utf-8` suffix.

```
$ curl -is http://localhost:4000/echo
HTTP/1.1 200 OK
X-Powered-By: Express
Cache-Control: max-age=300
Content-Type: application/json; charset=utf-8
Content-Length: 52
ETag: W/"34-xd75nzJDCHWlG7oIv/EAD/4iR1s"
Date: Mon, 31 Jul 2023 07:28:58 GMT
Connection: keep-alive

{"name":"rasel","age":25,"sub":["bangla","english"]}
```

And [Kong's document](https://docs.konghq.com/hub/kong-inc/proxy-cache/configuration/) also has this hint:

> content_type
> ...
> Note that `*/*` does not match with a content type that contains a parameter (e.g. `*/*` would not match `application/json; charset=utf-8`). If the response content type does not match the configuration, a `Bypass` cache status is returned.

So you need to change the plugin's configuration `content_type` to `application/json; charset=utf-8` and that would work!

**Reply by raselSoftNode** [reply]
i tried like this. but cache is not work

```
plugins:
- name: proxy-cache
  config:
    response_code:
    - 200
    request_method:
    - GET
    - HEAD
    content_type:
    - application/json; charset=utf-8
    cache_ttl: 300
    strategy: memory
    cache_control: false
```

**Reply by ms2008** [reply]
Try enclosing `application/json; charset=utf-8` in double-quotes. Then curl a couple more times. It's working after I change the plugin configuration with a request like the following (I'm using traditional mode)

```
$ curl -is -X PATCH http://localhost:8001/plugins/3341ccdd-8c74-40f7-9334-871a3e62649c \
   --data "config.content_type=application/json; charset=utf-8"
```
